src/middlewared/middlewared/plugins/system.py

```diff
@@ -152,6 +155,323 @@ def debug(self, job):
+                )
+
+        ip_addresses = await self.middleware.call(
+            'interfaces.ip_in_use'
+        )
+        ip4_addresses_list = [alias_dict['address'] for alias_dict in ip_addresses if alias_dict['type'] == 'INET']
+        ip6_addresses_list = [alias_dict['address'] for alias_dict in ip_addresses if alias_dict['type'] == 'INET6']
+
+        ip4_address = data.get('ui_address')
+        if (
+            ip4_address and
+            ip4_address != '0.0.0.0' and
+            ip4_address not in ip4_addresses_list
+        ):
+            verrors.add(
+                f'{schema}.guiaddress',
+                'Selected ipv4 address is not associated with this machine'
+            )
+
+        ip6_address = data.get('ui_v6address')
+        if (
+            ip6_address and
+            ip6_address != '::' and
+            ip6_address not in ip6_addresses_list
+        ):
+            verrors.add(
+                f'{schema}.guiv6address',
+                'Selected ipv6 address is not associated with this machine'
+            )
+
+        syslog_level = data.get('sysloglevel')
+        if not syslog_level:
+            verrors.add(
+                f'{schema}.sysloglevel',
+                'This field is required'
+            )
+
+        syslog_server = data.get('syslogserver')
+        if not syslog_server:
+            verrors.add(
+                f'{schema}.syslogserver',
+                'This field is required'
+            )
+        else:
+            match = re.match("^[\w\.\-]+(\:\d+)?$", syslog_server)
+            if not match:
+                verrors.add(
+                    f'{schema}.syslogserver',
+                    'Invalid syslog server format'
+                )
+            elif ':' in syslog_server:
+                port = int(syslog_server.split(':')[-1])
+                if port < 0 or port > 65535:
+                    verrors.add(
+                        f'{schema}.syslogserver',
+                        'Port specified should be between 0 - 65535'
+                    )
+
+        protocol = data.get('ui_protocol')
+        if not protocol:
+            verrors.add(
+                f'{schema}.guiprotocol',
+                'This field is required'
+            )
+        else:
+            if protocol != 'http':
+                certificate_id = data.get('ui_certificate')
+                if not certificate_id:
+                    verrors.add(
+                        f'{schema}.guicertificate',
+                        'Protocol has been selected as HTTPS, certificate is required'
+                    )
+                else:
+                    # getting fingerprint for certificate
+                    fingerprint = await self.middleware.call(
+                        'certificate.get_fingerprint',
+                        certificate_id
+                    )
+                    if fingerprint:
+                        syslog.openlog(logoption=syslog.LOG_PID, facility=syslog.LOG_USER)
```

Comment:
@william-gr is there any reason `syslog` is used here instead of `self.logger`?

Reply:
@themylogin yes, we need it to be logged to /var/log/messages and console to keep old funcionality.

Now that you have mentioned, LOG_USER is probably not the right facility.

Reply:
I mean, we could set up a syslog handler and use logger, but thats a whole lot of work to tackle in this PR ; )